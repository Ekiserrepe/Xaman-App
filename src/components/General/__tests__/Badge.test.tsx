import React from 'react';

import renderer from 'react-test-renderer';

import { AppColors } from '@theme';

import { TouchableDebounce } from '../TouchableDebounce';
import { Badge, BadgeType } from '../Badge';

describe('[Badge]', () => {
    it('renders correctly with default props', () => {
        const tree = renderer.create(<Badge />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with a label', () => {
        const tree = renderer.create(<Badge label="Test Label" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with specified type', () => {
        const tree = renderer.create(<Badge type={BadgeType.Xrplns} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies correct color based on the type', () => {
        const tree = renderer.create(<Badge type={BadgeType.Xrplns} />);
        expect(tree.toJSON()).toMatchSnapshot();
        const backgroundColorStyle = tree.root.findByType(TouchableDebounce).props.style[1].backgroundColor;
        expect(backgroundColorStyle).toEqual(AppColors.brandXrplns);
    });

    it('calls onPress callback when pressed', () => {
        const onPressMock = jest.fn();
        const tree = renderer.create(<Badge onPress={onPressMock} />);
        tree.root.findByType(TouchableDebounce).props.onPress();
        expect(onPressMock).toHaveBeenCalled();
    });

    it('renders correctly with different sizes', () => {
        const treeSmall = renderer.create(<Badge size="small" />).toJSON();
        expect(treeSmall).toMatchSnapshot();

        const treeMedium = renderer.create(<Badge size="medium" />).toJSON();
        expect(treeMedium).toMatchSnapshot();

        const treeLarge = renderer.create(<Badge size="large" />).toJSON();
        expect(treeLarge).toMatchSnapshot();
    });
});
