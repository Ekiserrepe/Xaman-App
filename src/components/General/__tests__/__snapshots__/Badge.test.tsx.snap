// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Badge] applies correct color based on the type 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#3767CE",
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "ProximaNova-Bold",
          "fontSize": 18,
        },
        {
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    XRPLNS
  </Text>
</View>
`;

exports[`[Badge] renders correctly with a label 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "ProximaNova-Bold",
          "fontSize": 18,
        },
        {
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    Test Label
  </Text>
</View>
`;

exports[`[Badge] renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
/>
`;

exports[`[Badge] renders correctly with different sizes 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
/>
`;

exports[`[Badge] renders correctly with different sizes 2`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
/>
`;

exports[`[Badge] renders correctly with different sizes 3`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
/>
`;

exports[`[Badge] renders correctly with specified type 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#3767CE",
      "borderRadius": 5,
      "marginLeft": 5,
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 1,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "ProximaNova-Bold",
          "fontSize": 18,
        },
        {
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    XRPLNS
  </Text>
</View>
`;
